### PR TITLE
Fix the KeyError issue after the environment_checker() is running

### PR DESCRIPTION
**Pull request by RazTamir**

### CODE REVIEW

**Comment by RazTamir** [comment]
E               ocs.exceptions.ResourceLeftoversException: 
E               There are leftovers in the environment after test case:
E               Resources added: [{'pvs': {'apiVersion': 'v1', 'kind': 'PersistentVolume', 'metadata': {'annotations': {'pv.kubernetes.io/provisioned-by': 'rbd.csi.ceph.com'}, 'creationTimestamp': '2019-06-25T12:44:46Z', 'finalizers': ['kubernetes.io/pv-protection'], 'name': 'pvc-ff68fefb-9746-11e9-aaed-0200b0aa1b0c', 'resourceVersion': '27636', 'selfLink': '/api/v1/persistentvolumes/pvc-ff68fefb-9746-11e9-aaed-0200b0aa1b0c', 'uid': '03269d16-9747-11e9-b5ca-06f322fbde0a'}, 'spec': {'accessModes': ['ReadWriteOnce'], 'capacity': {'storage': '3Gi'}, 'claimRef': {'apiVersion': 'v1', 'kind': 'PersistentVolumeClaim', 'name': 'test-pvc-re81', 'namespace': 'test-project', 'resourceVersion': '25332', 'uid': 'ff68fefb-9746-11e9-aaed-0200b0aa1b0c'}, 'csi': {'driver': 'rbd.csi.ceph.com', 'fsType': 'ext4', 'nodePublishSecretRef': {'name': 'secret-test-2515373740', 'namespace': 'openshift-storage'}, 'volumeAttributes': {'adminid': 'admin', 'clusterID': 'openshift-storage', 'imageFeatures': 'layering', 'imageFormat': '2', 'pool': 'cbp-test-2515373911', 'storage.kubernetes.io/csiProvisionerIdentity': '1561465260642-8081-'}, 'volumeHandle': '0001-0011-openshift-storage-0000000000000004-0217e48f-9747-11e9-b8ee-0a580a80020c'}, 'persistentVolumeReclaimPolicy': 'Delete', 'storageClassName': 'storageclass-test-2515374449', 'volumeMode': 'Filesystem'}, 'status': {'phase': 'Released'}}}]
E               Resources removed: []

utility/environment_check.py:144: ResourceLeftoversException

**Comment by fbalak** [comment]
I tried it with fresh installation but test fails with following errors in pytest log:
```
(...)
17:47:50 - MainThread - utility.utils - INFO - Executing command: for i in `seq 1 100`;do oc delete pvc test-pvc-re$i -n test-project;done
Error from server (NotFound): persistentvolumeclaims "test-pvc-re1" not found
Error from server (NotFound): persistentvolumeclaims "test-pvc-re2" not found
Error from server (NotFound): persistentvolumeclaims "test-pvc-re3" not found
(...)
Error from server (NotFound): persistentvolumeclaims "test-pvc-re98" not found
Error from server (NotFound): persistentvolumeclaims "test-pvc-re99" not found
Error from server (NotFound): persistentvolumeclaims "test-pvc-re100" not found
17:49:12 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/ocsqe/data/ocs-instances/current/auth/kubeconfig get StorageClass storageclass-test-2517445872 -o yam
(...)
```
And in traceback I see:
```

request = <SubRequest 'test_fixture' for <Function test_multiple_pvc_concurrent_creation_deletion>>

    @pytest.fixture()
    def test_fixture(request):
        """
        Setup and teardown
        """
        self = request.node.cls
    
        def finalizer():
            teardown(self)
        request.addfinalizer(finalizer)
>       setup(self)

tests/test_pvc_concurrent_deletion_creation.py:32: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'tests.test_pvc_concurrent_deletion_creation.TestMultiplePvcConcurrentDeletionCreation'>

    def setup(self):
        """
        Create new project
        Create PVCs
        """
        # Create new project
        assert PROJECT.new_project(TEST_PROJECT), (
            f'Failed to create new project {TEST_PROJECT}'
        )
    
        # Parameters for PVC yaml as dict
        pvc_data = load_yaml_to_dict(constants.CSI_PVC_YAML)
        pvc_data['metadata']['namespace'] = TEST_PROJECT
        pvc_data['spec']['storageClassName'] = self.sc_obj.name
        pvc_data['metadata']['name'] = self.pvc_base_name
    
        # Create 100 PVCs
        pvc_objs = create_multiple_pvc(self.number_of_pvc, pvc_data)
        log.info(f'Created initial {self.number_of_pvc} PVCs')
        self.pvc_objs_initial = pvc_objs[:]
    
        # Verify PVCs are Bound and fetch PV names
        for pvc in self.pvc_objs_initial:
            pvc.reload()
>           assert pvc.status == constants.STATUS_BOUND, (
                f'PVC {pvc.name} is not Bound'
            )
E           AssertionError: PVC test-pvc1 is not Bound
E           assert 'Pending' == 'Bound'
E             - Pending
E             + Bound

tests/test_pvc_concurrent_deletion_creation.py:59: AssertionError
```

File: utility/environment_check.py [hunk 1]
```diff
@@ -140,18 +140,28 @@ def get_status_after_execution():
     leftovers = {'Leftovers added': [], 'Leftovers removed': []}
     for kind, kind_diff in diffs_dict.items():
         if ADDED_RESOURCE in kind_diff:
-            leftovers['Leftovers added'].append({
-                kind: kind_diff[ADDED_RESOURCE][
-                    ''.join(kind_diff[ADDED_RESOURCE])
-                ]
-            })
+            try:
+                leftovers['Leftovers added'].append({
```
Comment (clacroix12) [inline utility/environment_check.py]: @RazTamir Do you have an example of what this ends up looking like once items are appended? The logic here around what we're appending isn't super clear. I know you haven't changed it here in this PR but I'm just curious if we could simplify it at all.

Comment (RazTamir) [reply utility/environment_check.py]: sure:
http://pastebin.test.redhat.com/774998
In that dictionary, you have a key named 'iterable_item_added' or 'iterable_item_removed' that includes a dict or list of more dictionaries that represent the added/removed items

**Comment by fbalak** [comment]
I encountered the problem from https://github.com/red-hat-storage/ocs-ci/pull/292#issuecomment-505509418 during testing of `tests/test_pvc_concurrent_deletion_creation.py` again.

**Comment by RazTamir** [comment]
@jilju this looks like a bug to me

**Comment by jilju** [comment]
> @jilju this looks like a bug to me

@RazTamir 
The error in [PR 292 (comment)](https://github.com/red-hat-storage/ocs-ci/pull/292#issuecomment-505509418) shows that the PVC is not in Bound state. That could be a real bug.
But I am confused here because I have executed this test many times but never encountered this issue.

**Comment by RazTamir** [comment]
@jilju maybe a race condition?
@fbalak could you please hsare with Jilju more info so he will be able to debug?

**Comment by fbalak** [comment]
> @fbalak could you please hsare with Jilju more info so he will be able to debug?

I sent @jilju a PM on chat.

**Comment by RazTamir** [comment]
All, could you please review this?

**Comment by jilju** [comment]
> > @fbalak could you please hsare with Jilju more info so he will be able to debug?
> 
> I sent @jilju a PM on chat.

@fbalak. Thank you. I will check

**Pull request merged**